tests/Integration/Behaviour/Features/Context/Domain/AliasFeatureContext.php

```diff
@@ -47,140 +46,104 @@
+        try {
+            $this->getCommandBus()->handle(new AddAliasCommand($aliases, $searchTerm));
+        } catch (InvalidArgumentException|AliasConstraintException $exception) {
+            $this->setLastException($exception);
         }
     }
 
     /**
-     * @When /^I (enable|disable) alias with reference "(.+)"$/
-     *
-     * @param bool $enable
-     * @param string $aliasReference
+     * @When I update search term :searchTerm with following aliases:
      *
-     * @see StringToBoolTransformContext::transformTruthyStringToBoolean for $enable string to bool transformation
+     * @param TableNode $table
      */
-    public function updateAliasStatus(bool $enable, string $aliasReference): void
+    public function updateSearchTermAliases(string $searchTerm, TableNode $table): void
     {
-        $this->getCommandBus()->handle(new UpdateAliasStatusCommand(
-            $this->getSharedStorage()->get($aliasReference),
-            $enable
-        ));
+        $this->editSearchTermCommand($searchTerm, $table);
     }
 
     /**
-     * @When /^I bulk (enable|disable) the following aliases "([^"]*)"$/
+     * @When I update search term :oldSearchTerm by :newSearchTerm with following aliases:
      *
-     * @param bool $status
-     * @param string $aliasReferences
+     * @param TableNode $table
      */
-    public function bulkUpdateStatusForDefaultShop(bool $status, string $aliasReferences): void
+    public function updateWithNewSearchTermAliases(string $oldSearchTerm, string $newSearchTerm, TableNode $table): void
     {
-        try {
-            $this->getCommandBus()->handle(new BulkUpdateAliasStatusCommand($this->referencesToIds($aliasReferences), $status));
-        } catch (AliasException $e) {
-            $this->setLastException($e);
-        }
+        $this->editSearchTermCommand($oldSearchTerm, $table, $newSearchTerm);
     }
 
     /**
-     * @Then I should get error that alias cannot be empty
-     * @Then I should get error that search term cannot be empty
+     * @Then following aliases should exist:
      *
-     * @return void
+     * @param TableNode $table
      */
-    public function assertLastErrorIsInvalidAliasConstraint(): void
+    public function assertAliasesList(TableNode $table): void
     {
-        $this->assertLastErrorIs(InvalidArgumentException::class);
-    }
+        $expectedAliases = $table->getColumnsHash();
 
-    /**
-     * @When I update alias ":aliasReference" with following values:
-     *
-     * @return void
-     */
-    public function assertUpdateAlias(string $aliasReference, TableNode $table): void
-    {
-        $updateAliasCommand = $this->buildUpdateAliasCommand($aliasReference, $table);
+        /** @var AliasQueryBuilder $aliasQueryBuilder */
+        $aliasQueryBuilder = $this->getContainer()->get(AliasQueryBuilder::class);
+        $aliases = $aliasQueryBuilder
+            ->getAliasQueryBuilder(new Filters([
```

Comment:
I'm not sure you need to make this method public to do this, marking a method public just for test purposes is not really a good practice I wonder if you could use the public `getSearchQueryBuilder` instead, and modify the return builder by adding the select you need? (I don't know if you can remove the groupBy though but maybe it's not needed)

This would prevent turning this private method into a public one

Reply:
Besides there's another issue you don't use the `GetAliasesBySearchTermForEditing` in this context meaning it's not validated by behat tests You should refacto this assertion step to make it validate one search term at a time, which still allows you to assert the created/updated aliases, it will validate the query as well and it prevents relying on the query builder and turning a private method into a public one